e2e_tests/tests/experiment/test_core.py

```diff
@@ -476,3 +476,14 @@ def test_max_concurrent_trials(name: str, searcher_cfg: str) -> None:
 
     finally:
         exp.cancel_single(experiment_id)
+
+
+@pytest.mark.e2e_cpu
+def test_core_api_non_conforming_steps_vals_checkpoints():
+    exp.run_basic_test(
```

Comment:
Should this test query the workloads or do something to make sure a query that was once affected is not affected anymore?

Reply:
sorry, i forgot to fix up the test, did most of this locally but then didn't translate it to the test. will fix and request re-review